client/src/components/events/CreateEventTypeForm.vue

```diff
@@ -210,14 +220,43 @@ export default {
     },
 
     computed: {
-        eventLinkDashed() {
-            return this.eventName.replace(/\s/g, '-');
+        ...mapGetters('eventType', {
+            getEventTypeForm: eventTypeGetters.GET_EVENT_TYPE_FORM
+        }),
+        data() {
+            return Object.keys(this.form).some(el => !!this.form[el]) ? this.form : this.getEventTypeForm;
         }
     },
 
     methods: {
+        ...mapMutations('eventType', {
+            setEventTypeForm: eventTypeMutations.SET_EVENT_TYPE_FORM
+        }),
+        clickNext() {
+            this.setEventTypeForm(this.form);
+        },
         setColor(id) {
-            this.color = this.colorById[id].id;
+            this.form.color = this.colorById[id].id;
+        },
+        changeName(val) {
+            this.form.name = val;
+        },
+        changeLocation(val) {
+            this.form.location = val;
+        },
+        changeDescription(val) {
+            this.form.description = val;
+        },
+        changeSlug(val) {
+            this.form.slug = val;
+        }
+    },
+
+    watch: {
+        'data.name': function(val, prevVal) {
+            if (prevVal === this.form.slug || this.form.slug.length < 1) {
```

Comment:
in `changeName` method call `changeSlug(value.replace(...))`. Also, do not use contractions, there is no tax on characters (val => value)